app/src/common/shared/com/igalia/wolvic/ui/widgets/WindowWidget.java

```diff
@@ -1877,6 +1877,9 @@ public void onFullscreen(@NonNull final WSession session, @NonNull final WMediaS
     public void onPageStart(@NonNull WSession aSession, @NonNull String aUri) {
         mCaptureOnPageStop = true;
         mViewModel.setIsLoading(true);
+        // Pages that are completely loaded from cache don't trigger onFirstContentfulPaint so we
+        // force it here to the get page properly rendered.
+        onFirstContentfulPaint(aSession);
```

Comment:
Hmm that's right, perhaps we could switch to a progress handler? I mean invoke it as soon as progress > 0 ?

Reply:
It's an old memory, but I've seen a case where simple pages went straight to PageLoadingStop from progress = 0. not sure if the case is still happened. Maybe adding both progress > 10 (?) & PageLoadingStop would be also possible.

Reply:
I've updated a new patch. I don't think we need to check progress > X because there are 2 possibilities:
1. FCP was not emitted: we issue it
2. FCP was emitted: then our call is a no-op

The problem with page loading stop is that it can be too late, in the sense that we won't get any rendering until the page is fully loaded